[MGDOBR-1344] - IstioSetupService should check if the expected resour…

vaibhavjainwiz opened the pull request:
…ce matches the existing one and createOrReplace it in case

(https://issues.redhat.com/browse/MGDOBR-1344)
Implement equal and hashcode methods for Istio POJOs

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

danielezonca reviewed (commented):
Can you add some test for `IstioSetupService` class?

vaibhavjainwiz commented:
> IstioSetupService

We had added `@UnlessBuildProfile(value = "test")` annotation on this class to skip from testing because it have `OnStartUp` event lister. 
We can't write unit testcases for it.

danielezonca commented:
> > IstioSetupService
> 
> We had added `@UnlessBuildProfile(value = "test")` annotation on this class to skip from testing because it have `OnStartUp` event lister. We can't write unit testcases for it.

That annotation makes that class not loaded/initialised by CDI during tests but it should be possible to write a unit tests. For example you can make `create*()` methods `protected` and invoke them explicitly.

vaibhavjainwiz commented:
> > > IstioSetupService
> > 
> > 
> > We had added `@UnlessBuildProfile(value = "test")` annotation on this class to skip from testing because it have `OnStartUp` event lister. We can't write unit testcases for it.
> 
> That annotation makes that class not loaded/initialised by CDI during tests but it should be possible to write a unit tests. For example you can make `create*()` methods `protected` and invoke them explicitly.

Hi @danielezonca 
I had added the testcases for this class by making its method protected. I also had to add an constructor to initialize its object in test class.

robpblake reviewed (approved):
Thanks @vaibhavjainwiz

vaibhavjainwiz commented on shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/IstioSetupServiceTest.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    @Test
+    public void test_gateway_modification_applied() {
```
done
